### PR TITLE
spectra/convolve.py: barak convolution routines replaced by astropy.c…

### PR DESCRIPTION
**Pull request by ntejos**
spectra/convolve.py: barak convolution routines replaced by astropy.convolution equivalents.

### CODE REVIEW

**Comment by ntejos** [comment]
It seems that the new implementation failed test_gauss_smooth() due to a difference of ~1e-3 whereas a precision closer to 1e-7 was required. If Neil and JXP are fine with me updating the test_gauss_smooth() function to pass with the new convolution implementation I will do so.

Also, we should consider to deprecate our concolve.py module and simply use astropy.convolution directly instead.

File: linetools/spectra/convolve.py [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
 """ 
 
 # py2.6+ compatibility
 from __future__ import print_function, absolute_import, division, unicode_literals
 
 import numpy as np
+from astropy.convolution import convolve, Gaussian1DKernel, CustomKernel
 
-def convolve_psf(a, fwhm, edge='invert', replace_nan=False, debug=False):
+def convolve_psf_old(a, fwhm, edge='invert', replace_nan=False, debug=False):
```
Comment (nhmc) [inline linetools/spectra/convolve.py]: I think it's better to remove the old versions completely (`convolve_psf_old` and `convolve_window_old`).

File: linetools/spectra/convolve.py [hunk 2]
```diff
@@ -1,14 +1,16 @@
 """ Functions related to convolution.
       Taken from Barak by JXP
       May replace with scipy functions
+      NT: replace barak routines with astropy.convolvolution equivalents
```
Comment (nhmc) [inline linetools/spectra/convolve.py]: Typo

**Comment by nhmc** [comment]
Apart from the inline comments, this looks great to me.  Thanks!  Feel to merge when you're ready.

**Pull request merged**